testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py

```diff
@@ -0,0 +1,193 @@
+                admin_client.delete_user_if_exists(sgs[key]["db"], sgs[key]["user"])
+            if (pre_test_db_exists is not None) and (pre_test_db_exists is False):
+                if admin_client.does_db_exist(sgs[key]["db"]) is True:
+                    admin_client.delete_db(sgs[key]["db"])
+
+
+@pytest.mark.syncgateway
+@pytest.mark.collections
+def test_scopes_and_collections_replication(scopes_collections_tests_fixture, params_from_base_test_setup):
+    """
+        @summary:
+        # 1. Create users, user sessions
+        # 2. Upload documents to the pushing SGW and make sure that there are no document on the passive SGW
+        # 3. Start a pull continous replication sg1 <- sg2
+        # 4. Upload new documents to sgw1
+        # 5. Add another collection
+        # 6. Start a push replication
+         # 7. Check that the new documents were replicated to sgw2
+    """
+    sg1 = sgs["sg1"]
+    sg2 = sgs["sg2"]
+    admin_client_1 = Admin(sg1["sg_obj"])
+    admin_client_2 = Admin(sg2["sg_obj"])
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+    num_of_docs = 3
+
+    pull_replication_prefix = "should_be_in_sg2_after_pull"
+    push_replication_prefix = "should_be_in_sg2_after_push"
+    sg_client, scope, collection, collection2 = scopes_collections_tests_fixture
+
+    if sync_gateway_version < "3.1.0":
+        pytest.skip('This test cannot run with Sync Gateway version below 3.1.0')
+
+    sg_client = MobileRestClient()
+
+    # 1. Create users, user sessions
+    password = "password"
+    user1_auth = sgs["sg1"]["user"], password
+    user2_auth = sgs["sg2"]["user"], password
+
+    # 2. Upload documents to the pushing SGW and make sure that there are no document on the passive SGW
+    uploaded_for_pull = sg_client.add_docs(url=sg1_url, db=sg1["db"], number=num_of_docs, id_prefix=pull_replication_prefix, channels=["A"], auth=user1_auth, scope=scope, collection=collection)
+    with pytest.raises(Exception) as e:
+        sg_client.get_doc(sg2_url, sg2["db"], uploaded_for_pull[0]["id"], rev=uploaded_for_pull[0]["rev"], auth=user2_auth, scope=scope, collection=collection)
+    e.match("Not Found")
+
+    # 3. start a pull continous replication sg1 <- sg2
+    replicator2_id = sg2["sg_obj"].start_replication2(
+        local_db=sg2["db"],
+        remote_url=sg1["sg_obj"].url,
+        remote_db=sg1["db"],
+        remote_user=sg1["user"],
+        remote_password=password,
+        direction="pull",
+        continuous=False,
+        collections_enabled=True
+    )
+    admin_client_2.wait_until_sgw_replication_done(sg2["db"], replicator2_id, read_flag=True, max_times=3000)
+
+    # 3. Check that the documents were replicated to sgw2
+    for i in range(0, num_of_docs - 1):
+        sg_client.get_doc(sg2_url, sg2["db"], uploaded_for_pull[i]["id"], rev=uploaded_for_pull[i]["rev"], auth=user2_auth, scope=scope, collection=collection)
+
+    # 4. Upload new documents to sgw1
```

Comment:
is it worth having the test have documents in both the previous collection and the newly added collection?

Reply:
Yes, it's worth, good idea :) Fixed.